crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -75,36 +84,57 @@ impl OperationLogs {
     }
 
     pub fn remove_outdated_logs(&self) -> Result<(), OperationLogsError> {
-        let mut update_logs = BinaryHeap::new();
-        let mut list_logs = BinaryHeap::new();
+        let mut log_tracker: HashMap<String, BinaryHeap<Reverse<String>>> = HashMap::new();
+
+        // FIXME: this is a hotfix to map "software-list" and "software-update" to "software-management"
+        // this should be fixed in https://github.com/thin-edge/thin-edge.io/issues/1077
         for file in (self.log_dir.read_dir()?).flatten() {
             if let Some(path) = file.path().file_name().and_then(|name| name.to_str()) {
-                if path.starts_with(UPDATE_PREFIX) {
-                    // The paths are pushed using the reverse alphabetic order
-                    update_logs.push(Reverse(path.to_string()));
-                } else if path.starts_with(LIST_PREFIX) {
-                    // The paths are pushed using the reverse alphabetic order
-                    list_logs.push(Reverse(path.to_string()));
+                if path.starts_with("software-list") {
+                    log_tracker
+                        .entry("software-list".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else if path.starts_with("software-update") {
+                    log_tracker
+                        .entry("software-update".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else {
+                    let file_name = path
+                        .split('-')
```

Comment:
Doesn't this logic expect that the operation name wouldn't have `-` in it? Would this logic break if my custom operation name was `my-custom-operation`?

Reply:
@albinsuresh, yes it does but remember we give the operation name in #1077 in the log plugin toml file

There is no point fixing it here, because we will need to change it again.